source/extensions/transformers/aws_lambda/api_gateway_transformer.cc

```diff
@@ -0,0 +1,200 @@
+  if (json_body.contains("statusCode")) {
+    uint64_t status_value;
+    try {
+      status_value = json_body["statusCode"].get<uint64_t>();
+    } catch (std::exception& exception){
+      ENVOY_STREAM_LOG(debug, "Error parsing statusCode: " + std::string(exception.what()), stream_filter_callbacks);
+      ApiGatewayError error = {500, "500", "Non-integer status code"};
+      return ApiGatewayTransformer::format_error(*response_headers, body, error, stream_filter_callbacks);
+    }
+    response_headers->setStatus(status_value);
+  } else {
+    response_headers->setStatus(DEFAULT_STATUS_VALUE);
+  }
+
+  // set response headers
+  if (json_body.contains("headers")) {
+    const auto& headers = json_body["headers"];
+    if (!headers.is_object()) {
+        ENVOY_STREAM_LOG(debug, "invalid headers object", stream_filter_callbacks);
+        ApiGatewayError error = {500, "500", "invalid headers object"};
+        return ApiGatewayTransformer::format_error(*response_headers, body, error, stream_filter_callbacks);
+    }
+    for (json::const_iterator it = headers.cbegin(); it != headers.cend(); it++) {
+        const auto& header_key = it.key();
+        const auto& header_value = it.value();
+        std::string header_value_string;
+        if (header_value.is_string()) {
+          header_value_string = header_value.get<std::string>();
+        } else {
+          header_value_string = it.value().dump();
+        }
+        add_response_header(*response_headers, header_key, header_value_string, stream_filter_callbacks, false);
+    }
+  }
+
+  // set multi-value response headers
+  if (json_body.contains("multiValueHeaders")) {
+    const auto& multi_value_headers = json_body["multiValueHeaders"];
+    if (!multi_value_headers.is_object()) {
+        ENVOY_STREAM_LOG(debug, "invalid multi headers object", stream_filter_callbacks);
+        ApiGatewayError error = {500, "500", "invalid multi headers object"};
+        return ApiGatewayTransformer::format_error(*response_headers, body, error, stream_filter_callbacks);
+    }
+
+    for (json::const_iterator it = multi_value_headers.cbegin(); it != multi_value_headers.cend(); it++) {
+        const auto& header_key = it.key();
+        const auto& header_values = it.value();
+
+        for (json::const_iterator inner_it = header_values.cbegin(); inner_it != header_values.cend(); inner_it++) {
+          const auto& header_value = inner_it.value().get<std::string>();
+          add_response_header(*response_headers, header_key, header_value, stream_filter_callbacks, true);
+        }
+    }
+  }
+
+  // set response body
+  body.drain(body.length());
+  if (json_body.contains("body")) {
+    std::string body_dump;
+    if (json_body["body"].is_string()) {
+      body_dump = json_body["body"].get<std::string>();
+    } else {
+      body_dump = json_body["body"].dump();
+    }
+    if (json_body.contains("isBase64Encoded") && json_body["isBase64Encoded"] == true) {
```

Comment:
we should check if `json_body["isBase64Encoded"]` is a bool type and then get / cast to bool before comparing